File: python/pycommons/math/comparison.py
"""Number comparison utilities"""

# A very small number used in floating-point comparison
VERY_SMALL_NUMBER = 1e-12


def epsilonEquals(a: float, b: float, epsilon: float) -> bool:
    """Check if two values are within a range of eachother

    Args:
        a (float): Value a
        b (float): Value b
        epsilon (float): Range to check in 

    Returns:
        bool: Are they near eachother?
    """

    return (a - epsilon <= b) and (a + epsilon >= b)

File: java/src/main/java/ca/retrylife/commons/math/Comparison.java
package ca.retrylife.commons.math;

public class Comparison {

    // A very small number to be used for comparing floats
    public static final double kVerySmallNumber = 1e-12;

    public static boolean epsilonEquals(int a, int b, int epsilon) {
        return (a - epsilon <= b) && (a + epsilon >= b);
    }

    public static boolean epsilonEquals(double a, double b) {
        return epsilonEquals(a, b, kVerySmallNumber);
    }

    public static boolean epsilonEquals(double a, double b, double epsilon) {
        return (a - epsilon <= b) && (a + epsilon >= b);
    }

    public static boolean epsilonEquals(float a, float b) {
        return epsilonEquals(a, b, kVerySmallNumber);
    }

    public static boolean epsilonEquals(float a, float b, float epsilon) {
        return (a - epsilon <= b) && (a + epsilon >= b);
    }

}